lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2186,7 +2233,32 @@ void FIRRTLLowering::initializeRegister(Value reg) {
     auto intType = type.cast<IntegerType>();
     if (intType.getWidth() == 0)
       return;
-    builder.create<sv::BPAssignOp>(dest, getRandomValue(intType));
+
+    SmallVector<SymbolAndRange> values;
+    getRandomValues(intType, values);
+
+    if (values.size() == 1) {
+      auto [value, range] = values[0];
+      builder.create<sv::VerbatimOp>(
+          builder.getStringAttr(Twine("{{0}} = {{1}}[") + Twine(range.first) +
+                                ":" + Twine(range.second) + "];"),
```

Comment:
Pretty please change this to generate `x[0]` instead of `x[0:0]`.

Reply:
Good suggestion (and this aligns with how ExportVerilog works). This little stuff really makes a difference in the output quality. 😄 

Updated in the force-push to do:
1. `foo[high]` if high == low
2. `foo` if high == 31, low == 0
3. `foo[high:low]` otherwise

Reply:
Thank you!